api.planx.uk/modules/analytics/metabase/shared/client.test.ts

```diff
@@ -1 +1,117 @@
-test.todo("should test configuration and errors");
+import axios from "axios";
+import {
+  validateConfig,
+  createMetabaseClient,
+  MetabaseError,
+} from "./client.js";
+import nock from "nock";
+
+const axiosCreateSpy = vi.spyOn(axios, "create");
+
+describe("Metabase client", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.resetModules();
+  });
+
+  afterEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  test("returns configured client", async () => {
+    const client = createMetabaseClient();
+    expect(client.defaults.headers["X-API-Key"]).toBe(
+      process.env.METABASE_API_KEY,
```

Comment:
It's worth testing our other setup vars here also like the baseURL, timeout and content-type 👍